### ISSUE TITLE
Format JSON, JSONCompact is not consistent with JSONEachRow response

### ISSUE THREAD

**Issue by doom369** [ClickHouse/ClickHouse#7814]
While working with `JSON`, `JSONCompact` and `JSONEachRow` I found some inconsistency between those 3 formats.

`JSON` and `JSONCompact` always return additional fields: `meta`, `rows`, `statistics`, `data`. Example of `JSONCompact` response:

```
{
	"meta":
	[
		{
			"name": "created",
			"type": "DateTime"
		},
		{
			"name": "value",
			"type": "Int32"
		}
	],

	"data":
	[
		["2019-11-17 11:31:22", 1],
		["2019-11-17 11:31:22", 2]
	],

	"rows": 2,

	"statistics":
	{
		"elapsed": 0.000745078,
		"rows_read": 2,
		"bytes_read": 16
	}
}
```

`JSONEachRow` returns the data without those fields:

```
{"created":"2019-11-17 11:31:22","value":1}
{"created":"2019-11-17 11:31:22","value":2}
```

I think for `JSON` and `JSONCompact` formats it would be correct to return the data without any additional fields, like `JSONEachRow` does:

```
	[
		["2019-11-17 11:31:22", 1],
		["2019-11-17 11:31:22", 2]
	]
```

```
        [
		{
			"value": 1,
                        "created":"2019-11-17 11:31:22"
		},
		{
			"value": 2,
                       "created":"2019-11-17 11:31:22"
		}
	]
```

At the moment there is no way to exclude `meta`, `rows`, `statistics` fields.
One of the most useful use cases for `JSON` format is to perform the request and send the response directly to the UI without any operations/transformations in the middleware. Forwarding of those meta field is not desired.

WDYT?

**Comment by tavplubix** [comment]
It's documented feature: https://clickhouse.yandex/docs/en/interfaces/formats/#json

`JSON` and `JSONCompact` formats output some additional information, which may be useful. `JSONEachRow` outputs each row as a separate JSON object without any additional information. You can use `JSONEachRow` if you don't need those fields. Possibly, we could add a setting to exclude meta information from `JSON` and `JSONCompact`, but it's impossible to determine names of columns in `"data"` without meta information, so I think it's better to use `JSONEachRow`, which outputs fields in `"column_name": value` format.

**Comment by doom369** [comment]
@tavplubix thanks. Yes, I read the documentation. Still I think it is not consistent. From my point of view all json formats should return the additional meta or all json format should not return the additional data.  Right now it is a mix. So I think it should be fixed. It is not big deal. Just want to point this out.

**Comment by den-crane** [comment]
output_format_write_statistics

cl --output_format_write_statistics=0 -q "select 1 format JSON"

**Comment by doom369** [comment]
@den-crane thanks, but is hides only 1 field. So problem remains.

**Comment by alexey-milovidov** [comment]
It can be implemented with two independent settings:
- `output_format_json_each_row_pretty_print` for indentation and whitespaces;
- `output_format_json_each_row_enclose_in_array` to enclose everything in single array;

**Comment by alexey-milovidov** [comment]
Implemented by @akuzm here: #17152
